### PR TITLE
SAIC-329 - applied unique VM names and internal net names for each build

### PR DESCRIPTION
**Pull request by andybondar**

### CODE REVIEW

**Comment by roman-verchikov** [comment]
1. Please squash commits into one
2. Keep in mind that `Vagrantfile` will also be used on people's workstations, so it should work out of the box if anyone wants to deploy devlab on a laptop. This means we should keep working `Vagrantfile` in `devlab` folder.

**Comment by andybondar** [comment]
> Please squash commits into one
Ok

> Keep in mind that Vagrantfile will also be used on people's workstations
Good point. That's why I don't touch devlab/Vagranfile in the repo and do all modifications in devlab/jenkins/Vagranfile.template, and rewrite devlab/Vagranfile when it's necessary.

**Comment by andybondar** [comment]
> Please squash commits into one

Done

File: devlab/jenkins/nosetests.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+cf_hostname=`vagrant status | grep running | grep cloudferry | awk '{print $1}'`
+
+echo "Copy code archive to cloudferry-${cf_hostname} VM ..."
+cf_ip=`vagrant ssh-config ${cf_hostname} | grep HostName | awk '{print $2}'`
+cf_user=`vagrant ssh-config ${cf_hostname} | grep -w "User" | awk '{print $2}'`
+cf_port=`vagrant ssh-config ${cf_hostname} | grep Port | awk '{print $2}'`
+cf_id=`vagrant ssh-config ${cf_hostname} | grep IdentityFile | awk '{print $2}'`
+
+
+#/home/vagrant/cloudferry/devlab/tests/nosetests.xml
```
Comment (roman-verchikov) [inline devlab/jenkins/nosetests.sh]: Debug leftover, please remove

File: devlab/jenkins/nosetests.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+cf_ssh_cmd="ssh -q -oConnectTimeout=5 -oStrictHostKeyChecking=no -oCheckHostIP=no -i ${cf_id} ${cf_user}@${cf_ip} -p ${cf_port}"
+run_nosetests="cloudferry/devlab/jenkins/cf/run_nosetests.sh"
+${cf_ssh_cmd} ${run_nosetests}
+
+scp -q -oConnectTimeout=5 -oStrictHostKeyChecking=no -oCheckHostIP=no -i ${cf_id} -P ${cf_port} ${cf_user}@${cf_ip}:cloudferry/devlab/tests/nosetests.xml ${CF_DIR}/devlab/tests/nosetests.xml
```
Comment (roman-verchikov) [inline devlab/jenkins/nosetests.sh]: Please make it multiline, it's very difficult to read

File: devlab/utils/os_cli.sh [hunk 3]
```diff
@@ -24,8 +26,28 @@ usage() {
...
+dst_user=`vagrant ssh-config $dst_hostname | grep -w "User" | awk '{print $2}'`
+
+src_port=`vagrant ssh-config $src_hostname | grep Port | awk '{print $2}'`
+dst_port=`vagrant ssh-config $dst_hostname | grep Port | awk '{print $2}'`
+
+src_id=`vagrant ssh-config $src_hostname | grep IdentityFile | awk '{print $2}'`
+dst_id=`vagrant ssh-config $dst_hostname | grep IdentityFile | awk '{print $2}'`
+
+#dst_ssh_cmd="ssh -o stricthostkeychecking=no root@${dst}"
+#src_ssh_cmd="ssh -o stricthostkeychecking=no root@${src}"
```
Comment (roman-verchikov) [inline devlab/utils/os_cli.sh]: Is this needed?.. Please remove if not

**Comment by roman-verchikov** [comment]
Looks good overall, but has one major issue: once this is merged, we'd have to maintain **two** versions of vagrant files: one for devlab, and the other for jenkins. This will definitely have an impact in the future. Not sure if it's possible to have both using the same codebase though...

**Comment by andybondar** [comment]
Looks like I've found approach how to modify Vagrantfile in order to have one universal version for both devlab ('manual' testing without unique hostnames) and jenkins (with unique hostnames for each build). I'll update request as soon as fix is ready.

**Comment by andybondar** [comment]
Pull request has been updated

**Comment by realsystem** [comment]
+1

**Comment by roman-verchikov** [comment]
Tested locally, works as charm. Code looks good too. Can be merged.

**Pull request merged**